marklogic-data-hub/src/main/java/com/marklogic/hub/step/RunStepResponse.java

```diff
@@ -90,6 +92,7 @@ public RunStepResponse withStep(String stepNum) {
         if (targetEntityTextNode != null) {
             this.targetEntityType = targetEntityTextNode.asText();
         }
+        Optional.of(step.getOptions()).map(optionMap -> (TextNode) optionMap.get("targetDatabase")).ifPresent(node -> this.targetDatabase = node.asText());
```

Comment:
It's a bit difficult to look at this and be sure there's no error. I think some unit tests are needed for the withStep method - none exist yet. Those tests would verify that targetDatabase is set if it exists in the options for the given step, but not set if targetDatabase doesn't exist in the given step's options but does exist in another step's options.